### ISSUE TITLE
Non-sargable rule logging error with built-in date/time functions with no column reference.

### ISSUE THREAD

**Issue by ltylenda** [tsqllint/tsqllint#314]
Non-sargable rule logging error with built-in date/time functions with no column reference.
The example below is sargable:
```
SELECT [Name] FROM Foo WHERE Foo.DateCreated >= DATEADD(HOUR, -4, GETDATE());
```
EXPECTED: No errors logged.
ACTUAL: A non-sargable error is logged.
